hw/ip/usbdev/data/usbdev_testplan.hjson

```diff
@@ -18,12 +19,795 @@
+              packets to the device.
+            - Don't service the packet reception by popping out from rx_fifo.
+            - Verify that this interrupt pin goes high once the available fifo is empty.
+            '''
+      stage: V1
+      tests: ["usbdev_av_empty"]
+    }
+    {
+      name: rx_full
+      desc: '''
+            Verify the functionality of rx FIFO when rx_full signal is high and device interface is enabled.
+
+            - Enable the interrupt for RX full by setting INTR_ENABLE[8] and then send back to back packets
+              to the device.
+            - Don't just service the packets received by popping out from rx_fifo.
+            - Verify that this interrupt pin should go high once the received fifo is full.
+            '''
+      stage: V1
+      tests: ["usbdev_rx_full"]
+    }
+    {
+      name: av_overflow
+      desc: '''
+            Verify that if a write was done to the Available Buffer FIFO when the FIFO was full then
+            av_overflow signal is high.
+
+            - Write a particular buffer ID to the available buffer FIFO when it is already full.
+            - Verify that this interrupt pin goes high indicating that the available fifo has
+              reached the overflowing condition.
+            '''
+      stage: V1
+      tests: ["usbdev_av_overflow"]
+    }
+    {
+      name: enable
+      desc: '''
+            Verify that USB connect to interface when enable is set.
+
+            - set USB_CTRL_enable pin high.
+            - Make sure phy_config_pinflip isn't set.
+            - Verify that the usb_dp_pullup_o pin goes high.
+            '''
+      stage: V1
+      tests: ["usbdev_enable"]
+    }
+    {
+      name: resume_link_active
+      desc: '''
+            Verify that when Write a 1 to this bit to instruct usbdev to jump to the LinkResuming state
+            then write will only have an effect when the device is in the LinkPowered state.
+
+            - Perform no operation in J state for more than 3 ms.
+            - Read the link state from usbstate register to see if it is in powered state and then send the
+              resume signal (K state) for 20 ms and set the usbctrl.resume_link_active.
+            - Verify that the resume_link_active interrupt goes high and device will be resumed.
+            '''
+      stage: V1
+      tests: ["usbdev_resume_link_active"]
+    }
+    {
+      name: device_address
+      desc: '''
+            Verify that device address set by host (this should be copied from the Set Device ID SETUP packet).
+            This will be zeroed by the hardware when the bus resets.
+
+            - read USB_CTRL[22:16] send an In token packet to the device on an enabled endpoint with read value
+              as the device address.
+            - Verify that if Device send any data packet in response then the read value is the actual address of the device.
```

Comment:
I think this description needs to be clearer, please.

What I think we're trying to check here is:
1. the assigned device address (ISP_CTRL{22:16]) may be changed by the DV environment on the CSR side.
2. the device responds to all packets sent by the DV environment (USB side) that are sent to its assigned address.
3. the device ignores all packets that are sent to other addresses.